src/main/kotlin/no/nav/familie/ba/sak/kjerne/brev/mottaker/BrevmottakerService.kt

```diff
@@ -64,53 +65,63 @@ class BrevmottakerService(
-    ): List<MottakerInfo> =
-        brevMottakere.map { brevmottaker ->
-            when (brevmottaker.type) {
-                MottakerType.FULLMEKTIG, MottakerType.VERGE -> {
-                    val finnesBrevmottakerMedUtenlandskAdresse =
-                        brevMottakere.any { it.type == MottakerType.BRUKER_MED_UTENLANDSK_ADRESSE }
-                    if (finnesBrevmottakerMedUtenlandskAdresse) { // brev sendes til fullmektig adresse og bruker sin manuell adresse
-                        MottakerInfo(
-                            brukerId = søkersident,
-                            brukerIdType = BrukerIdType.FNR,
-                            erInstitusjonVerge = false,
-                            navn = brevmottaker.navn,
-                            manuellAdresseInfo = lagManuellAdresseInfo(brevmottaker),
-                        ).toList()
-                    } else { // brev sendes til fullmektig adresse og bruker sin registerte adresse
-                        listOf(
-                            MottakerInfo(
-                                brukerId = søkersident,
-                                brukerIdType = BrukerIdType.FNR,
-                                erInstitusjonVerge = false,
-                                navn = brevmottaker.navn,
-                                manuellAdresseInfo = lagManuellAdresseInfo(brevmottaker),
-                            ),
-                            MottakerInfo(
-                                brukerId = søkersident,
-                                brukerIdType = BrukerIdType.FNR,
-                                erInstitusjonVerge = false,
-                                navn = søkersnavn,
-                            ),
-                        )
-                    }
-                }
-
-                MottakerType.BRUKER_MED_UTENLANDSK_ADRESSE, MottakerType.DØDSBO ->
-                    // brev sendes til kun bruker sin registerte/manuell adresse
-                    MottakerInfo(
-                        brukerId = søkersident,
-                        brukerIdType = BrukerIdType.FNR,
-                        erInstitusjonVerge = false,
-                        navn = søkersnavn,
-                        manuellAdresseInfo = lagManuellAdresseInfo(brevmottaker),
-                    ).toList()
+    ): List<MottakerInfo> {
+        val manuellDødsbo = manueltRegistrerteMottakere.filter { it.type == MottakerType.DØDSBO }
+            .map {
+                MottakerInfo(
+                    brukerId = "",
+                    brukerIdType = null,
+                    erInstitusjonVerge = false,
+                    navn = søkersnavn,
+                    manuellAdresseInfo = lagManuellAdresseInfo(it),
+                )
+            }.singleOrNull()
+
+        if (manuellDødsbo != null) {
+            // brev sendes kun til den manuelt registerte dødsboadressen
+            return manuellDødsbo.toList()
+        }
```

Comment:
Siden det kastes feil senere dersom det eksisterer flere enn 1 manuelldødsbo, og man egentlig bare returnerer hvis det er 1, kan man ikke bare ta singleOrNull sjekken først?

Istedenfor å mappe potensielt flere addresser, også senere kaste feil.

Noe ala

```suggestion
        val manuellDødsbo = manueltRegistrerteMottakere.filter { it.type == MottakerType.DØDSBO }.singleOrNull()?.let {
            MottakerInfo(
                brukerId = "",
                brukerIdType = null,
                erInstitusjonVerge = false,
                navn = søkersnavn,
                manuellAdresseInfo = lagManuellAdresseInfo(it),
            )
        }

        if (manuellDødsbo != null) {
            // brev sendes kun til den manuelt registerte dødsboadressen
            return manuellDødsbo.toList()
        }
```

Reply:
Jo, da blir det mer likt det kodemønsteret som det er endret til under også 👍

Reply:
Lager en oppfølgings PR med diverse utbedringer etter denne. Prioriterer å rette feilen i prod først..